strawberry/asgi/__init__.py

```diff
@@ -54,6 +54,11 @@ async def get_root_value(self, request: Request) -> typing.Optional[typing.Any]:
     async def get_context(self, request: Request) -> typing.Optional[typing.Any]:
         return {"request": request}
 
+    async def get_websocket_context(
+        self, websocket: WebSocket
```

Comment:
Since both the `Request` and `Websocket` objects are representations of a request (and have the same API for accessing request details like headers and query params), I'd recommend we stick with one context generator function instead of two.

Having both `get_context` and `get_websocket_context` as methods that users are expected to override means more effort for users. It also exposes an implementation detail of subscriptions (since theoretically they are protocol agnostic). 

Instead, we could modify `get_context` to accept a union type of `Union[Request, Websocket]` as the `request` parameter. That way, users only have to override one method to set the context for both websocket and http connections, and they can access request details using the same API for both.

Reply:
I like that! maybe in future we can also have an abstraction so we expose only one type that has the same API (if possible)

Reply:
+1 on keeping a single method `get_context` that accepts a request or a websocket